### DISCUSSION TITLE
travel guide creation - all lat/lon=0 ?!

### DISCUSSION THREAD

**Discussion by mariush444** [osmandapp/OsmAnd#13208] [Q&A]
I try to create simple travel guide but no success.

I have `guide.html` (just one header and one line) and `guide.gpx` with one point in the guide directory/folder. [guide.zip](https://github.com/osmandapp/OsmAnd/files/7570768/guide.zip)
Then I started command `./utilities.sh travel-guide-creator` from OsmandMapCreator v.3.0

log of creation [creation.zip](https://github.com/osmandapp/OsmAnd/files/7570667/creation.zip)
 
The file [guide.travel.obf.zip](https://github.com/osmandapp/OsmAnd/files/7570685/guide.travel.obf.zip) was copied by me to osmand-folder/`travel` directory. 

I can see simple page and one point in Osmand but ....
the point has lat/lon=0

[guide.gpx.zip](https://github.com/osmandapp/OsmAnd/files/7570711/guide.gpx.zip) that I found in osmand-folder/`tracks/travel`

```
  <wpt lat="0.00" lon="0.00">
    <name>1st point</name>
    <type>Other</type>
    <extensions>
      <osmand:color>#d00d0d</osmand:color>
      <osmand:icon>special_star</osmand:icon>
    </extensions>
  </wpt>
```
_additionally the points is doubled in the gpx file._

When I tried to create guide with some numbers of points then every point got lat/lon=0.

[inspection.zip](https://github.com/osmandapp/OsmAnd/files/7570930/inspection.zip) - result of the command `./utilities.sh inspector -vpoi -v guide.travel.obf `

I have no idea what/how to investigate more. Currently travel guid is not sqlite but obf so ...

What I do in wrong way during creation?

**Comment by vshcherb** [comment]
Created an issue based on it https://github.com/osmandapp/OsmAnd/issues/13211.

**Comment by Dima-1** [comment]
I checked creation travel guide with your files, works as expected
[test.travel.obf.zip](https://github.com/osmandapp/OsmAnd/files/7572668/test.travel.obf.zip)
[log.zip](https://github.com/osmandapp/OsmAnd/files/7572669/log.zip)

**Comment by mariush444** [comment]
hmmm....

Where is the problem then?
Java?
I use 
```
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7)
OpenJDK 64-Bit Server VM (build 16.0.2+7, mixed mode)
```
I noticed that original gpx file contains
`  <wpt lat="52.4851835" lon="20.3185089">`
but gpx file that I found in your test.travel.obf
contains two point with
`  <wpt lat="52.4851846" lon="20.3185058">`

Was any conversion done in meantime?

**Comment by mariush444** [comment]
I switched to Linux mint 20.1 cinnamon (generally I work on Manjaro)
with
```
openjdk 11.0.11 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
```
and I can confirm that guide was created OK.

The only problem for this moment with this simple file is that the POI is doubled.

I compared creation log files but I can't find any special problems.

(diff only)
![image](https://user-images.githubusercontent.com/66887280/142739371-81db7877-8987-4caf-969a-6c634902f01a.png)

**Reply by vshcherb** [reply]
Weird... probably related to old Map Creator version?

**Reply by mariush444** [reply]
both (on Manjaro and Mint - shared exctracted folder) are the same v.3.0.
https://download.osmand.net/latest-night-build/OsmAndMapCreator-main.zip

When I know that travel guide is gpx file with embeded html then it is no such important for me.
I can create it semi-manualy.
You (your team) can swith to other/more important issues then.

PS
for me, it would be nice to have possibility to use local storage images on smartphone (e.g. osmand_folder/avnotes or DCIM - if there will be problem with future google rules.)
`<img "src=picture.jpg">` insted `<img "src=https://website.com/picture.jpg">`

**Reply by mariush444** [reply]
I found reason
I had LC_`CTYPE=pl_PL.UTF-8`
when I switched to `CTYPE=` 
then travel guide is created OK.

Still 2 POI's but the lat/lon is OK.

```
routes: route_article guide Lat 52.485184 Lon 20.318506 osmid=-1000  route_name:en='guide' route_id='Q1' description:en='gzip ...' name:en='guide'
routes: route_article_point 1st point Lat 52.485184 Lon 20.318506 osmid=-1001  category_other='other' lang_yes:en='yes' route_name='guide' route_id='Q1' gpx_bg='circle' color='red' gpx_icon='special_star'
routes: route_article_point 1st point Lat 52.485184 Lon 20.318506 osmid=-1002  category_other='other' lang_yes:en='yes' route_name='guide' route_id='Q1' gpx_bg='circle' color='red' gpx_icon='special_star'
```
